core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -202,25 +213,39 @@ public void ping(PingCallback callback, Executor executor) {
     return new CallImpl<ReqT, RespT>(method, new SerializingExecutor(executor));
   }
 
-  private synchronized ClientTransport obtainActiveTransport() {
-    if (shutdown) {
-      return null;
-    }
-    if (activeTransport != null) {
-      return activeTransport;
+  private ClientTransport obtainActiveTransport() {
+    ClientTransport savedActiveTransport = activeTransport;
```

Comment:
nit: could you combine `savedActiveTransport` and `newActiveTransport` and just call it `transport` so that it's always the same variable you return in all cases?